### PR TITLE
Feature/4779 4842 implement view model

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR adds the implementation of the view model. Creating the OTP-URL.
It also adds the new default app config.

## Link to Jira
[4842](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4842)

### CODE REVIEW

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyConsentViewController.swift [hunk 1]
```diff
@@ -40,14 +40,19 @@ final class SurveyConsentViewController: DynamicTableViewController, ENANavigati
...
-		guard let url = URL(string: "https://www.test.de") else {
-			return
+		navigationFooterItem?.isPrimaryButtonLoading = true
+
+		viewModel.getURL { [weak self] result in
+			switch result {
+			case .success(let url):
+				self?.completion(url)
+			case .failure(let error):
+				self?.showErrorAlert(with: error)
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyConsentViewController.swift:50]: Would be nice to log the error

Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyConsentViewController.swift:50]: Our does the Service already do this?

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift [hunk 2]
```diff
@@ -59,7 +59,7 @@ final class CachedAppConfiguration {
 
 	/// The location of the default app configuration.
 	private var defaultAppConfigPath: URL {
-		guard let url = Bundle.main.url(forResource: "default_app_config_18", withExtension: "") else {
+		guard let url = Bundle.main.url(forResource: "default_app_config_113", withExtension: "") else {
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift:62]: 👍

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift [hunk 3]
```diff
@@ -385,6 +385,14 @@ enum AppStrings {
 		static let legalDetailsButtonTitle = NSLocalizedString("SurveyConsent_Legal_Details_Button_Title", comment: "")
 		static let legalDetailsTitle = NSLocalizedString("SurveyConsent_Legal_Details_Title", comment: "")
 		static let legalDetailsBody = NSLocalizedString("SurveyConsent_Legal_Details_Body", comment: "")
+
+		// Errors
+		static let errorTitle = NSLocalizedString("SurveyConsent_Error_Title", comment: "")
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift:390]: String file got changed in an other PR?

**Review by marcussc — APPROVED** [review]

**Review by KaiTeuber — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyConsentViewController.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyConsentViewController.swift:50]: The logs should be at the service layers.

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift:390]: We have the strings on a separated branch.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionCoordinator.swift [hunk 4]
```diff
@@ -48,7 +54,24 @@ final class ExposureDetectionCoordinator {
...
-		let surveyConsentViewController = SurveyConsentViewController(viewModel: SurveyConsentViewModel()) { [weak self] url in
+		// ToDo: Replace with real services
+
+		let deviceCheckMock = PPACDeviceCheckMock(true, deviceToken: "iPhone")
+		guard let ppacService = try? PPACService(
+			store: store,
+			deviceCheck: deviceCheckMock
+		) else {
+			return
+		}
```
[the remark this answers is not in the document]
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionCoordinator.swift:65]: We will wire up the real services at the end.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyConsentViewModel.swift [hunk 5]
```diff
@@ -3,11 +3,72 @@
 //
 
 import UIKit
+import OpenCombine
+
+enum SurveyConsentError: Error {
+
```
[the remark this answers is not in the document]
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyConsentViewModel.swift:8]: done

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyConsentViewController.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyConsentViewController.swift:50]: I'll add them.

**Pull request merged**